src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Logging.Console.Internal
+{
+    public class ConsoleLoggerProcessor : IDisposable
+    {
+        private const int _maxQueuedMessages = 1024;
+
+        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+        private readonly Task _outputTask;
+
+        public IConsole Console;
+
+        public ConsoleLoggerProcessor()
+        {
+            // Start Console message queue processor
+            _outputTask = Task.Factory.StartNew(
```

Comment:
I think we should file an issue to avoid using a dedicated thread. It's probably fine for the most part but it can be done in a way where we don't have a thread spun up all the time. I don't know how much of a problem it is in practice though.